burn-wgpu/src/template/prng/uniform_int_inner_loop.wgsl

```diff
@@ -0,0 +1,18 @@
+let low = args[0];
+let high = args[1];
+let range = high - low;
+
+for (var i = 0u; i < n_values_per_thread; i++) {
+    let write_index = write_index_base + i * n_threads_per_workgroup;
+
+    state[0u] = taus_step_0(state[0u]);
+    state[1u] = taus_step_1(state[1u]);
+    state[2u] = taus_step_2(state[2u]);
+    state[3u] = lcg_step(state[3u]);
+    let random_u32 = state[0u] ^ state[1u] ^ state[2u] ^ state[3u];
+
+    // Modulus operation to fit within the range
+    let random_in_range = (random_u32 % range) + low;
+
+    output[write_index] = random_in_range;
+}
```

Comment:
@louisfd is this shader being invoked for the test that's failing? The problem on CI was a fusion test `fusion::base::tests::aggregation::tests::test_should_mean_last_dim_int`, that used to work at commit [ae60d25](https://github.com/tracel-ai/burn/pull/1136/commits/ae60d257f63163e1cefc48cd0e67e8b81309c199) but then stopped working at [dca5b26](https://github.com/tracel-ai/burn/pull/1136/commits/dca5b26ff6889996ed846058cf134c4933352887) for some reason. It works locally for me though.

Now it seems the CI is complaining about something else though.

Reply:
@louisfd Ok I'm back to failing on `test_should_mean_last_dim_int()` in the aggregation tests for burn-tensor. For both that and `test_should_sum_last_dim_int()`, shouldn't we be calling the `int_mean_dim` and `int_sum_dim()` though? Still not sure why `cast_float()` is causing issues though especially since I never use it, and if these tests use the base tensor functions, neither should even be calling anything from this PR.

```rust
#[test]
    fn test_should_mean_last_dim_int() {
        let tensor = TestTensorInt::from([[0, 1, 2], [3, 4, 5]]);

        let data_actual = tensor.mean_dim(1).to_data();

        assert_eq!(data_actual, Data::from([[1], [4]]));
    }

    #[test]
    fn test_should_sum_last_dim_int() {
        let tensor = TestTensorInt::from([[0, 1, 2], [3, 4, 5]]);

        let data_actual = tensor.sum_dim(1).to_data();

        assert_eq!(data_actual, Data::from([[3], [12]]));
    }
```